### ISSUE TITLE
Page scroll stuck on some position, infinite loading shown

### ISSUE THREAD

**Issue by DmitryPopov** [chrismccord/todo_trek#5]
Steps to reproduce:

- scroll page to bottom, with till text "You made it to the beginning of time"
- scroll page to top to "kanban" board
- page scroll stuck at some point, can't scroll top or bottom
 

https://github.com/chrismccord/todo_trek/assets/1150117/3dd04678-f7ba-4458-a037-73547eb4f2bc

**Comment by chrismccord** [comment]
do the server logs show errors when you get into that loading bar/scroll state?

**Comment by DmitryPopov** [comment]
@chrismccord Don't see any errors in logs, debug logs for handle events only.
```
[debug] HANDLE EVENT "prev-page" in TodoTrekWeb.HomeLive
  Parameters: %{"id" => "activity_logs-15"}
[debug] QUERY OK source="activity_log_entries" db=25.4ms queue=0.2ms idle=1961.8ms
SELECT a0."id", a0."meta", a0."action", a0."performer_text", a0."subject_text", a0."before_text", a0."after_text", a0."todo_id", a0."list_id", a0."user_id", a0."inserted_at", a0."updated_at" FROM "activity_log_entries" AS a0 WHERE (a0."user_id" = $1) ORDER BY a0."inserted_at" DESC LIMIT $2 OFFSET $3 [1, 20, 20]
↳ TodoTrekWeb.HomeLive.paginate_logs/2, at: lib/todo_trek_web/live/home_live.ex:183
[debug] Replied in 27ms
[debug] HANDLE EVENT "prev-page" in TodoTrekWeb.HomeLive
  Parameters: %{"id" => "activity_logs-23"}
[debug] QUERY OK source="activity_log_entries" db=2.0ms queue=0.5ms idle=1474.4ms
SELECT a0."id", a0."meta", a0."action", a0."performer_text", a0."subject_text", a0."before_text", a0."after_text", a0."todo_id", a0."list_id", a0."user_id", a0."inserted_at", a0."updated_at" FROM "activity_log_entries" AS a0 WHERE (a0."user_id" = $1) ORDER BY a0."inserted_at" DESC LIMIT $2 OFFSET $3 [1, 20, 0]
↳ TodoTrekWeb.HomeLive.paginate_logs/2, at: lib/todo_trek_web/live/home_live.ex:183
[debug] Replied in 3ms
```

**Comment by DmitryPopov** [comment]
@chrismccord I've found next error in the browser console logs
<img width="541" alt="Screenshot 2023-06-01 at 18 47 54" src="https://github.com/chrismccord/todo_trek/assets/1150117/e5df3d1d-5cef-4df3-8fbd-d29a7b94e278">

**Comment by chrismccord** [comment]
Any steps to reliably recreate this? I can't recreate it on my side in safari or chrome.

**Comment by chrismccord** [comment]
Have you modified the templates at all? That error would only be returned if the stream item's parentElement was null somehow, which I don't know how that could happen.

**Comment by DmitryPopov** [comment]
I’ve not changed any files, scrolled page using touchpad only on Mac.
Reproduced on Safari and Chrome.

**Comment by chrismccord** [comment]
Ive scrolled hundreds of times across browsers and haven't ever triggered this so I can't say for sure what is happening or how the parentElement of one of the `<li>`'s wouldn't exist.

**Comment by chrismccord** [comment]
I was able to recreate it. It can only be triggered when the total items is a certain size or something. `mix ecto.reset` on my side, plus trying to mark a task as complete a few times was able to get it to happen. Will see what's going on now!

**Comment by chrismccord** [comment]
This one was fun. So it turns out the pagination code I wrote orders the timeline in reverse chronological order (which is what we want), but using `inserted_at` form the seeds means we lacked the granularity on the initial dataset to properly paginate over the results because they all had the same inserted time. Coupled with various other conditions like the number of items and the size of the stream limit, means that LiveView in some cases would remove an existing stream item that it was also about to re-insert because the pagination code returned the same item(s) for multiple pages. The fix on this side is to properly paginate over the data, which by id is fine since we have an auto inc pkey. On the LiveView side I also will detect this to prevent it from happening. Thanks!
